Modules/_blake2/blake2s_impl.c

```diff
@@ -159,6 +159,10 @@ py_blake2s_new_impl(PyTypeObject *type, PyObject *data, int digest_size,
     if (leaf_size_obj != NULL) {
         leaf_size = PyLong_AsUnsignedLong(leaf_size_obj);
         if (leaf_size == (unsigned long) -1 && PyErr_Occurred()) {
+            if (PyErr_ExceptionMatches(PyExc_OverflowError)) {
+                PyErr_SetString(PyExc_OverflowError,
+                                "leaf_size must be between 0 and 2**32-1");
```

Comment:
This error message and error message at line 169 should be same.

It may be worth to raise ValueError for negative leaf_size. Are there other restrictions for upper limit? Since this is specific to BLAKE2b it is worth to open a separate issue about this.